### DISCUSSION TITLE
TextBox Placeholder - Is it possible to change its color?

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#9293] [Q&A] [answered]
So basically I have a login form and I want to change the color of the placeholder as well as set another color whenever it is focused because it is barely visible right now.
![image](https://user-images.githubusercontent.com/114802572/198325688-d829ae6d-c897-4b6b-88fc-1e93bdfb7f01.png)
Is it possible? If yes, how to do it?

**Comment by maxkatz6** [comment, marked as answer]
@eXmatrx yes, it is possible by styling a watermerk component inside of the textbox:
```xaml
<Style Selector="TextBox /template/ TextBlock#PART_Watermark">
    <Setter Property="Foreground" Value="Gray" />
</Style>
```
note that you might want to have a different style per:
`Selector="TextBox:pointerover /template/ TextBlock#PART_Watermark"`
`Selector="TextBox:pressed /template/ TextBlock#PART_Watermark"`
as well

**Reply by ghost** [reply]
Thanks man for answering, I appreciate!
